src/core/geometry/qgsgeometry.cpp

```diff
@@ -1849,6 +1886,17 @@ bool QgsGeometry::convertToMultiType()
     return true;
   }
 
+  // TriangulatedSurface does not inherit from QgsGeometryCollection
+  if ( QgsWkbTypes::flatType( d->geometry->wkbType() ) == Qgis::WkbType::Triangle )
```

Comment:
Again, we've got an inconsistency here -- calling convertToMultiType on a triangle will return a geometry which is NOT a multitype! :scream:

Reply:
Removed.